packages/react-components/src/components/trace-context-component.tsx

```diff
@@ -211,33 +212,23 @@ export class TraceContextComponent extends React.Component<TraceContextProps, Tr
     }
 
     private doHandleTimeGraphResetSignal() {
-        this.unitController.viewRange = { start: 0, end: this.unitController.absoluteRange };
+        this.unitController.viewRange = { start: BigInt(0), end: this.unitController.absoluteRange };
     }
 
     private zoomButton(zoomIn: boolean) {
-        let newStartRange = 0;
-        let newEndRange = 0;
-        let position = (this.unitController.viewRange.end + this.unitController.viewRange.start) / 2;
-        const percentZoom = 0.9;
+        let position = (this.unitController.viewRange.end + this.unitController.viewRange.start) / BigInt(2);
         if (this.unitController.selectionRange) {
-            const positionMiddleSelection = (this.unitController.selectionRange.end + this.unitController.selectionRange.start) / 2;
+            const positionMiddleSelection = (this.unitController.selectionRange.end + this.unitController.selectionRange.start) / BigInt(2);
             if (positionMiddleSelection >= this.unitController.viewRange.start && positionMiddleSelection <= this.unitController.viewRange.end) {
                 position = positionMiddleSelection;
             }
         }
-        const startDistance  = position - this.unitController.viewRange.start;
-        const endDistance = this.unitController.viewRange.end - position;
-        const zoomFactor = zoomIn ? percentZoom : 1 / percentZoom;
-        newStartRange = position - startDistance * zoomFactor;
-        newEndRange = position + endDistance * zoomFactor;
-        if (newStartRange < 0) {
-            newEndRange = newEndRange - newStartRange;
-        } else if (newEndRange > this.unitController.absoluteRange) {
-            const delta = newEndRange - this.unitController.absoluteRange;
-            newStartRange = newStartRange - delta;
-        }
-        newStartRange = Math.max(newStartRange, 0);
-        newEndRange = Math.min(newEndRange, this.unitController.absoluteRange);
+        const startDistance = position - this.unitController.viewRange.start;
+        const zoomFactor = zoomIn ? 0.8 : 1.25;
+        const newDuration = BIMath.clamp(Number(this.unitController.viewRangeLength) * zoomFactor,
+            BigInt(2), this.unitController.absoluteRange);
+        const newStartRange = BIMath.max(0, position - BIMath.round(Number(startDistance) * zoomFactor));
```

Comment:
Yes we do. My work assumption was that in user navigation it's OK for the 'change' to have a loss of precision. I didn't think it was avoidable due to floating-point zoom factor, but I guess it could be done with integer math as *5/4 and *4/5?

Reply:
Yes, but I thought it was acceptable because the zoom factor itself is not a value that needs to be exact, e.g. the user wouldn't care if we zoom by 24.998% instead of 25.000%, for example.

Reply:
Ok, I see.